pkg/agent/agent.go

```diff
@@ -1132,27 +1132,20 @@ func getRoundInfo(bridgeClient ovsconfig.OVSBridgeClient) types.RoundInfo {
 	return roundInfo
 }
 
-func (i *Initializer) getNodeMTU(transportInterface *net.Interface) (int, error) {
+func (i *Initializer) getInterfaceMTU(transportInterface *net.Interface) (int, error) {
 	if i.mtu != 0 {
 		return i.mtu, nil
 	}
 	mtu := transportInterface.MTU
-	// Make sure mtu is set on the interface.
+	// Make sure MTU is set on the interface.
 	if mtu <= 0 {
 		return 0, fmt.Errorf("Failed to fetch Node MTU : %v", mtu)
 	}
-	if i.networkConfig.TrafficEncapMode.SupportsEncap() {
-		if i.networkConfig.TunnelType == ovsconfig.VXLANTunnel {
-			mtu -= config.VXLANOverhead
-		} else if i.networkConfig.TunnelType == ovsconfig.GeneveTunnel {
-			mtu -= config.GeneveOverhead
-		} else if i.networkConfig.TunnelType == ovsconfig.GRETunnel {
-			mtu -= config.GREOverhead
-		}
-		if i.nodeConfig.NodeIPv6Addr != nil {
-			mtu -= config.IPv6ExtraOverhead
-		}
-	}
+
+	isIPv6 := i.nodeConfig.NodeIPv6Addr != nil
+	i.networkConfig.MTUDeduction = i.networkConfig.CalculateMTUDeduction(isIPv6)
```

Comment:
I feel CalculateMTUDeduction() should set `MTUDeduction` internally.

Reply:
Nit: seems no need to define a variable for isIPv6.

Reply:
Done,set MTUDeduction internally, ipIPv6 is still required since networkConfig.IPv6Enabled is unset yet when `getInterfaceMTU` is called.